### PR TITLE
fix: inngest example for latest version

### PR DESCRIPTION
**Pull request by joelhooks**
This updates the Inngest Next.js example to function with the latest version of the Inngest SDK and generally simplify the structure for demo of the tool.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: ad9e61a8244b22f188d1c15f4558fa2cb7d1273a

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
## Failing test suites 
Commit: ad9e61a8244b22f188d1c15f4558fa2cb7d1273a

`TURBOPACK=1 pnpm test-start test/e2e/next-test/next-test.test.ts` (turbopack)
- next test > first time setup > should correctly install missing dependencies and generate missing configuration file for first-time-setup-js
- next test > first time setup > should correctly install missing dependencies and generate missing configuration file for first-time-setup-ts

<details>
<summary>Expand output</summary>

  ● next test › first time setup › should correctly install missing dependencies and generate missing configuration file for first-time-setup-js

    expect(received).toContain(expected) // indexOf

    Expected substring: "Successfully generated playwright.config.js. Create your first test and then run `next experimental-test`."
    Received string:    "
    Installing devDependencies (yarn):
    - @playwright/test·
    yarn add v1.22.19
    info No lockfile found.
    [1/4] Resolving packages...
    [2/4] Fetching packages...
    info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
    "

      54 |           expect(stdout).toContain('Installing devDependencies')
      55 |           expect(stdout).toContain('@playwright/test')
    > 56 |           expect(stdout).toContain(
         |                          ^
      57 |             `Successfully generated ${
      58 |               fixtureName === 'first-time-setup-js'
      59 |                 ? 'playwright.config.js'

      at toContain (e2e/next-test/next-test.test.ts:56:26)

  ● next test › first time setup › should correctly install missing dependencies and generate missing configuration file for first-time-setup-ts

    expect(received).toContain(expected) // indexOf

    Expected substring: "Successfully generated playwright.config.ts. Create your first test and then run `next experimental-test`."
    Received string:    "
    Installing devDependencies (yarn):
    - @playwright/test·
    yarn add v1.22.19
    info No lockfile found.
    [1/4] Resolving packages...
    [2/4] Fetching packages...
    info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
    "

      54 |           expect(stdout).toContain('Installing devDependencies')
      55 |           expect(stdout).toContain('@playwright/test')
    > 56 |           expect(stdout).toContain(
         |                          ^
      57 |             `Successfully generated ${
      58 |               fixtureName === 'first-time-setup-js'
      59 |                 ? 'playwright.config.js'

      at toContain (e2e/next-test/next-test.test.ts:56:26)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by djfarrelly — APPROVED** [review]
I'm from the Inngest team and I have reviewed these changes for accuracy and updates.

Only suggestion @joelhooks is to pin the `inngest` dependency on `3.x` to avoid future breaking changes as well.

File: examples/inngest/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "@types/react-dom": "18.2.7",
     "concurrently": "^8.2.1",
     "encoding": "^0.1.13",
-    "typescript": "5.2.2"
+    "inngest-cli": "latest",
```
Comment (devjiwonchoi) [inline examples/inngest/package.json:22]: @djfarrelly Is it `inngest` or `inngest-cli`?
x-ref: https://www.npmjs.com/package/inngest

**Comment by devjiwonchoi** [comment]
@joelhooks The branch is targeted to `main`, should be `canary`.
x-ref: https://github.com/vercel/next.js/blob/main/contributing/core/developing.md

File: examples/inngest/package.json (hunk 1 above)
Comment (djfarrelly) [reply examples/inngest/package.json:22]: Yep! For additional clarification, the inngest package is the SDK and the inngest-cli is used to spin up the dev server which communicates with the Next.js application.

Comment (joelhooks) [reply examples/inngest/package.json:22]: I updated the `inngest` package above to `3.x` - I believe the CLI is correct at latest.

**Comment by joelhooks** [comment]
@devjiwonchoi this PR now properly targets canary

**Review by leerob — APPROVED** [review]

**Pull request merged**